linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/KnowledgeGuardian.scala

```diff
@@ -369,6 +360,88 @@ private[emitter] final class KnowledgeGuardian(config: CommonPhaseConfig) {
       fieldDefsAskers.clear()
     }
   }
+
+  private class SpecialInfo(initClassClass: Option[LinkedClass],
+      initHijackedClasses: Iterable[LinkedClass]) extends Unregisterable {
+
+    private var isParentDataAccessed =
+      computeIsParentDataAccessed(initClassClass)
+
+    private var methodsInHijackedClasses =
+      computeMethodsInHijackedClasses(initHijackedClasses)
+
+    private val methodsInHijackedClassesAskers = mutable.Set.empty[Invalidatable]
+
+    def update(classClass: Option[LinkedClass],
+        hijackedClasses: Iterable[LinkedClass]): Boolean = {
+      val newMethodsInHijackedClasses = computeMethodsInHijackedClasses(hijackedClasses)
+
+      if (newMethodsInHijackedClasses != methodsInHijackedClasses) {
+        methodsInHijackedClasses = newMethodsInHijackedClasses
+        invalidateAskers(methodsInHijackedClassesAskers)
+      }
+
+      val newIsParentDataAccessed = computeIsParentDataAccessed(classClass)
+
+      val invalidateAll = isParentDataAccessed != newIsParentDataAccessed
+
+      invalidateAll
+    }
+
+    private def computeIsParentDataAccessed(classClass: Option[LinkedClass]): Boolean = {
+      def methodExists(linkedClass: LinkedClass, methodName: MethodName): Boolean = {
+        linkedClass.methods.exists { m =>
+          m.value.flags.namespace == MemberNamespace.Public &&
+          m.value.methodName == methodName
+        }
+      }
+
+      classClass.exists(methodExists(_, getSuperclassMethodName))
+    }
+
+    private def computeMethodsInHijackedClasses(
+        hijackedClasses: Iterable[LinkedClass]): Set[(ClassName, MethodName)] = {
+      val pairs = for {
+        hijackedClass <- hijackedClasses
+        method <- hijackedClass.methods
+        if method.value.flags.namespace == MemberNamespace.Public
+      } yield {
+        (hijackedClass.className, method.value.methodName)
+      }
+
+      pairs.toSet
+    }
+
+    def askIsParentDataAccessed(invalidatable: Invalidatable): Boolean =
+      isParentDataAccessed
+
+    def askHijackedClassHasPublicMethod(invalidatable: Invalidatable,
+        className: ClassName, methodName: MethodName): Boolean = {
+      invalidatable.registeredTo(this)
+      methodsInHijackedClassesAskers += invalidatable
+      methodsInHijackedClasses.contains((className, methodName))
+    }
+
+    def unregister(invalidatable: Invalidatable): Unit = {
+      methodsInHijackedClassesAskers -= invalidatable
+    }
+
+    /** Call this when we invalidate all caches. */
+    def unregisterAll(): Unit = {
+      methodsInHijackedClassesAskers.clear()
+    }
+  }
+
+  private def invalidateAskers(askers: mutable.Set[Invalidatable]): Unit = {
+    /* Calling `invalidateAndUnregisterFromAll()` will cause the
```

Comment:
There seems to be a mismatch between `invalidateAndUnregisterFromAll` in the comment versus `invalidate` in the actual code. I know it was already like that before the change, but maybe use the opportunity to fix the inconsistency?

Reply:
Good catch.